### PR TITLE
Add feature class, and update feature Student Application

### PR DESCRIPTION
**Pull request by ishtiaqniloy**
SchoolClass, ClassDetails, ClassStudent, and StudentApplication models and corresponding repositories, controllers, services, DTOs, and tests have been added. Note that admins data, and teachers data are also added in this branch for testing purposes. If main is updated with the previous PRs, I will merge the changes.

### CODE REVIEW

**Review by nazmussaif-dsi — COMMENTED** [review]
In `src/main/java/com/ideal/studentlog/services/StudentApplicationService.java` -> `map()` -> `studentApplication.setApprovedBy(adminRepository.findById(dto.getApprovedBy()).orElseThrow());`
error is not thrown explicitly.

File: pom.xml [hunk 1]
```diff
@@ -95,12 +95,6 @@
 			<version>RELEASE</version>
 			<scope>compile</scope>
 		</dependency>
-		<dependency>
-			<groupId>org.jetbrains</groupId>
-			<artifactId>annotations</artifactId>
-			<version>RELEASE</version>
-			<scope>compile</scope>
-		</dependency>
```
Comment (Shuhin) [inline pom.xml:103]: If this change gets merged wouldn't other face problems?

Comment (ishtiaqniloy) [reply pom.xml:103]: the same dependency was added twice. So I removed one

**Comment by ishtiaqniloy** [comment]
> In `src/main/java/com/ideal/studentlog/services/StudentApplicationService.java` -> `map()` -> `studentApplication.setApprovedBy(adminRepository.findById(dto.getApprovedBy()).orElseThrow());`
> error is not thrown explicitly.

fixed this issue.

**Review by nazmussaif-dsi — APPROVED** [review]

**Pull request closed without merging**